src/java/org/apache/cassandra/guardrails/Guardrail.java

```diff
@@ -633,80 +622,116 @@ private void ensureUpToDate()
             {
                 // Setting cachedAllowed first so that on a parse failure we leave everything as it previously
                 // was (not that we'd expect that matter but ...).
-                cachedDisallowed = current.stream()
-                                          .map(parser)
-                                          .collect(Collectors.toCollection(HashSet::new));
+                cachedValues = current.stream()
+                                      .map(parser)
+                                      .collect(Collectors.toCollection(() -> new HashSet<>(current.size())));
+                                      // FIXME: Note DB-4909 used SetsFactory
+                                      // .collect(SetsFactory.collectorForSize(current.size()));
                 cachedRaw = current;
             }
             catch (Exception e)
             {
-                // This catches parsing errors. Hopefully, this shouldn't happen as guardrails settings should have
+                // This catch parsing errors. Hopefully, this shouldn't happen as guardrails settings should have
```

Comment:
catches was correct :)

Reply:
! no idea what happened here - will correct it